src/pudl/output/ferc714.py

```diff
@@ -0,0 +1,400 @@
+                names=["report_date", "respondent_id_ferc714"]
+            )
+        ).reset_index()
+    )
+    return pd.merge(rids_ferc714, dates_rids_df, on="respondent_id_ferc714")
+
+
+def get_all_utils(pudl_out):
+    """
+    Compile IDs and Names of all known EIA Utilities.
+
+    Grab all EIA utility names and IDs from both the EIA 861 Service Territory table and
+    the EIA 860 Utility entity table. This is a temporary function that's only needed
+    because we haven't integrated the EIA 861 information into the entity harvesting
+    process and PUDL database yet.
+
+    Returns:
+        pandas.DataFrame: Having 2 columns ``utility_id_eia`` and ``utility_name_eia``.
+
+    """
+    return (
+        pd.concat([
+            pudl_out.utils_eia860()[["utility_id_eia", "utility_name_eia"]],
+            pudl_out.service_territory_eia861()[["utility_id_eia", "utility_name_eia"]],
+        ]).dropna(subset=["utility_id_eia"]).drop_duplicates(subset=["utility_id_eia"])
+    )
+
+
+class FERC714Respondents(object):
+    """
+    A class coordinating compilation of data related to FERC 714 Respondents.
+
+    The FERC 714 Respondents themselves are not complex as they are reported, but
+    various ambiguities and the need to associate service territories with them mean
+    there are a lot of different derived aspects related to them which we repeatedly
+    need to compile in a self consistent way. This class allows you to choose several
+    parameters for that compilation, and then easily access the resulting derived
+    tabular outputs.
+
+    Some of these derived attributes are computationally expensive, and so they are
+    cached internally. You can force a new computation in most cases by using
+    ``update=True`` in the access methods. HOwever, this functionality isn't totally
+    implemented because we're still depending on the interim ETL processes for the FERC
+    714 and EIA 861 data, and we don't want to trigger whole new ETL runs every time
+    a derived value is updated.
+
+    Attributes:
+        pudl_out (pudl.output.pudltabl.PudlTabl): The PUDL output object which should be
+            used to obtain PUDL data.
+        pudl_settings (dict or None): A dictionary of settings indicating where data
+            related to PUDL can be found. Needed to obtain US Census DP1 data which
+            has the county geometries.
+        ba_ids (ordered collection or None): EIA IDs that should be treated as referring
+            to balancing authorities in respondent categorization process. If None, all
+            known values of ``balancing_authority_id_eia`` will be used.
+        util_ids (ordered collection or None): EIA IDs that should be treated as
+            referring to utilities in respondent categorization process. If None, all
+            known values of ``utility_id_eia`` will be used.
+        priority (str): Which type of entity should take priority in the categorization
```

Comment:
This is valid; I like Enums for situations where there are a fixed number of acceptable inputs.

Reply:
How do you specify an enumeration in the docstring / argument / attribute context? I don't seem to know what to google for.